CORE-8744 Align Cpi related entities and repository with latest JPA guidance

filipesoliveira opened the pull request:
File checksums and cpi signer summary hashes are now strongly typed. An instance of `SecureHash` is now used instead of `String` since the former provides more context to people.
`CpiIdentifier` type should be used in the methods instead of separate arguments. This helps keeping the list of arguments for a method small and provides more context.
Deleted `PersistenceUtils.kt` to simplify the code base. To help reduce complexity, variables should always have the same name across the code base and not variations of the original name. If someone requires a string version of a SecureHash, they can call the method `toString()`.
Other but small refactors were also made.

filipesoliveira commented:
This PR should only be merged after https://github.com/corda/corda-runtime-os/pull/3241.

owenstanford reviewed (commented):
LGTM, will need to resolve conflicts once #3241 goes in can approve after

vkolomeyko commented on testing/test-utilities/src/main/kotlin/net/corda/test/util/dsl/entities/cpx/CpiBuilder.kt:
```diff
@@ -111,12 +112,12 @@ class CpiBuilder(private val randomId: UUID = UUID.randomUUID()) {
         if (name == null) name = "name_$randomCpkId"
         if (version == null) version = "version_$randomCpkId"
-        if (signerSummaryHash == null) signerSummaryHash = "signerSummaryHash_$randomCpkId"
+        if (signerSummaryHash == null) signerSummaryHash = SecureHash("SHA1","signerSummaryHash_$randomCpkId".toByteArray())
```
Since `SHA1` is broken, it is probably not to use at all, but change to `SHA256` instead. This probably applies to many classes in this package.
Although I do appreciate that they are being used for testing purposes, but they my set a wrong example.

Another interesting point that KDoc for our `SecureHash` says "Provides utilities for generating a cryptographic hash using different algorithms (currently only SHA-256 supported)" :-)

filipesoliveira replied:
Ok, Viktor. That sounds sensible. I will make those changes. Hopefully, find and replace will make the process easy.

filipesoliveira committed:
CORE-8744 Align Cpi related entities and repository with latest JPA guidance

File checksums and cpi signer summary hashes are now strongly typed. An instance
of SecureHash is now used instead of String since the former provides more context to people.

CpiIdentifier type should be used in the methods instead of separate arguments.
This helps keeping the list of arguments for a method small.

Deleted PersistenceUtils.kt to simplify the code base. To help reduce complexity of the code,
variables should always have the same name across the code base and not variations of the
original name. If someone requires a string version of a SecureHash, they can call the method toString().

Other but small refactors were also made.

filipesoliveira committed:
Fix build issue after rebase.

filipesoliveira committed:
Replaced the "SHA1" hash algorithm for "SHA-256".

filipesoliveira committed:
Fixed issue after rebasing.
